# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/target/lvp/patches/xfree86_hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -pruN xc_orig/fonts/util/ucs2any.c xc/fonts/util/ucs2any.c
--- xc_orig/fonts/util/ucs2any.c	2003-09-21 12:54:14.000000000 +0200
+++ xc/fonts/util/ucs2any.c	2004-01-16 19:49:18.000000000 +0100
@@ -48,9 +48,6 @@
 #include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
-#ifndef NEED_BASENAME
-#include <libgen.h>
-#endif
 #include <limits.h>
 #include <stdarg.h>
 #include <stdio.h>
@@ -61,17 +58,6 @@
 /* global variable for argv[0] */
 const char *my_name = NULL;
 
-#ifdef NEED_BASENAME
-static char *
-basename(char *pathname)
-{
-	char	*ptr;
-
-	ptr = strrchr(pathname, '/');
-	return ((ptr == NULL) ? pathname : &ptr[1]);
-}
-#endif
-
 /* "CLASS" "z" string and memory manipulation */
 
 static void *
diff -pruN xc_orig/programs/glxgears/Imakefile xc/programs/glxgears/Imakefile
--- xc_orig/programs/glxgears/Imakefile	2001-04-02 23:12:11.000000000 +0200
+++ xc/programs/glxgears/Imakefile	2004-01-16 19:44:27.000000000 +0100
@@ -2,7 +2,7 @@ XCOMM $XConsortium: Imakefile /main/7 19
 XCOMM $XFree86: xc/programs/glxgears/Imakefile,v 1.1 2001/04/02 21:12:11 dawes Exp $
 #include <Threads.tmpl>
         DEPLIBS = $(DEPGLXLIB) $(DEPXLIB)
-LOCAL_LIBRARIES = $(GLXLIB) $(XLIB) 
+LOCAL_LIBRARIES = $(GLXLIB) $(XLIB) -ldl -lXxf86vm
   SYS_LIBRARIES = MathLibrary
 
 SimpleProgramTarget(glxgears)
diff -pruN xc_orig/programs/glxinfo/Imakefile xc/programs/glxinfo/Imakefile
--- xc_orig/programs/glxinfo/Imakefile	2002-11-22 23:56:03.000000000 +0100
+++ xc/programs/glxinfo/Imakefile	2004-01-16 19:41:59.000000000 +0100
@@ -5,13 +5,13 @@ XCOMM $XFree86: xc/programs/glxinfo/Imak
 #if BuildGLULibrary
 
         DEPLIBS = $(DEPGLULIB) $(DEPGLXLIB) $(DEPXLIB)
-LOCAL_LIBRARIES = $(GLULIB) $(GLXLIB) $(XLIB) 
+LOCAL_LIBRARIES = $(GLULIB) $(GLXLIB) $(XLIB) -ldl -lXxf86vm
         DEFINES = -DDO_GLU
 
 #else
 
         DEPLIBS = $(DEPGLXLIB) $(DEPXLIB)
-LOCAL_LIBRARIES = $(GLXLIB) $(XLIB) 
+LOCAL_LIBRARIES = $(GLXLIB) $(XLIB) -ldl -lXxf86vm
 
 #endif
 
diff -pruN xc_orig/programs/xdm/Imakefile xc/programs/xdm/Imakefile
--- xc_orig/programs/xdm/Imakefile	2003-12-19 05:51:15.000000000 +0100
+++ xc/programs/xdm/Imakefile	2004-01-16 18:50:57.000000000 +0100
@@ -121,16 +121,7 @@ SYS_LIBRARIES1 = -lauth
 SYS_LIBRARIES1 = -ls
 #endif
 
-#if HasLibCrypt
-#ifdef SpecialLibCrypt
-CRYPT_LIBRARIES = SpecialLibCrypt
-#else
 CRYPT_LIBRARIES = -lcrypt
-#if defined(LynxOSArchitecture)
-CRYPT_DEFINES = -DHAS_CRYPT
-#endif
-#endif
-#endif
 
 #if HasBSD44Sockets
 SOCK_DEFINES = -DBSD44SOCKETS
diff -pruN xc_orig/extras/ogl-sample/main/gfx/lib/glu/libnurbs/internals/mymath.h xc/extras/ogl-sample/main/gfx/lib/glu/libnurbs/internals/mymath.h
--- xc_orig/extras/ogl-sample/main/gfx/lib/glu/libnurbs/internals/mymath.h	2002-05-31 18:36:49.000000000 +0200
+++ xc/extras/ogl-sample/main/gfx/lib/glu/libnurbs/internals/mymath.h	2004-01-26 10:04:10.000000000 +0100
@@ -66,8 +66,8 @@ extern "C" float	floorf(float);
 #define NEEDCEILF
 #endif
 #elif defined(NEEDCEILF)
-extern "C" float	ceilf(float);
-extern "C" float	floorf(float);
+//extern "C" float	ceilf(float);
+//extern "C" float	floorf(float);
 #endif
 
 #ifdef LIBRARYBUILD
